peachjam/models/generic_document.py

```diff
@@ -55,6 +55,16 @@ class Legislation(CoreDocument):
     class Meta:
         verbose_name_plural = "Legislation"
 
+    @property
+    def get_taxonomies(self):
```

Comment:
Since taxonomies can apply to any object, I think it would be  better to put this onto CoreDocument.